src/main/java/TasksList.java

```diff
@@ -0,0 +1,132 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class TasksList {
+    private Integer noOfTasks;
+    private List<Task> list = new ArrayList<Task>();
+
+    public TasksList(){
+        this.noOfTasks = 0;
+    }
+
+    /**
+     * Add task T into TaskList
+     * @param t
+     */
+    public void addTask(Task t) {
+        this.noOfTasks++;
+        this.list.add(t);
+        System.out.println("Hai! You task is added ~");
+        System.out.print("    ");
+        t.printTask();
+        System.out.println("You have " + this.noOfTasks + " tasks remaining");
+    }
+
+    /**
+     * list out the Tasks in the TaskList
+     */
+    public void show() {
+        System.out.println("Osu! Your task is as follows:");
+        for(int i = 0; i < this.noOfTasks; i++) {
+            System.out.print("  ");
+            System.out.print((i + 1) + ".");
+            this.list.get(i).printTask();
+        }
+    }
+
+    /**
+     * Find task with description String s (case-sensitive), and mark it as done.
+     * @param s
+     */
+    public void markAsDone(String s) throws SalmonNotInListException{
+        Integer index = this.findIndexWithDesc(s);
+        if (index == -1) {
+            throw new SalmonNotInListException();
+        } else {
+            this.list.get(index).markAsDone();
+        }
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.print("    ");
+        this.list.get(index).printTask();
+    }
+
+    /**
+     * Mark task at number index as done
+     * @param index
+     */
+    public void markAsDone(Integer index) throws SalmonNotInListException{
+        index -= 1;
+        if (index < 0 || index > (this.noOfTasks - 1)) {
+            throw new SalmonNotInListException();
+        } else {
+            this.list.get(index).markAsDone();
+        }
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.print("    ");
+        this.list.get(index).printTask();
+    }
+
+    /**
+     * Find task with description String s (case-sensitive), and mark it as not done.
+     * @param s
+     */
+    public void markAsNotDone(String s) throws SalmonNotInListException{
+        Integer index = this.findIndexWithDesc(s);
+        if (index == -1) {
+            throw new SalmonNotInListException();
+        } else {
+            this.list.get(index).markAsNotDone();
+        }
+        System.out.println("As you wished, your task is now undone:");
+        System.out.print("    ");
+        this.list.get(index).printTask();
+    }
+
+    /**
+     * Mark task at number index as not done
+     * @param index
+     */
+    public void markAsNotDone(Integer index) throws SalmonNotInListException{
+        index -= 1;
+        if (index < 0 || index > (this.noOfTasks - 1)) {
```

Comment:
This is a complicated expression as per coding standards. You might want to break this statement down into two boolean variables, example - ((isIndexLessThanZero) || (Same for RHS) ) to ensure the code is easily understandable for the reader.